Molecular Consequence as a Statement - schema and examples

mbrush opened the pull request:
This PR proposes a model for representing Molecular Consequence Annotations as Statements (as an alternative to a StudyResult-based model in #336). It is based on prior work by the VA team, community feedback from the user community (e.g. Sarah Hunt, Ensembl/VEP), and discussions with Epic developers (Mrinal and Katherine).

The PR includes:
- addition of a `VariantMolecularConsequenceProposition` class in va-core-source.yaml.
- addition of an `ancillaryInfo` property on the core Statement class in va-core-source.yaml. (to support capture of ancillary info VEP provides alongside the central MC annotation)
- VA-Spec compliant data examples for [transcript-consequence](https://github.com/ga4gh/va-spec/blob/molecular-consequence-statement-and-examples/tests/fixtures/vep-transcript-consequence-statement.yaml) and [intergenic-consequence](https://github.com/ga4gh/va-spec/blob/molecular-consequence-statement-and-examples/tests/fixtures/vep-intergenic-consequence-statement.yaml) statements in the tests/fixtures directory.

The document [here](https://docs.google.com/document/d/1DWRPjopZcVpyp1zMvjh7J3yplUIJc3R4E26COJok6Cw/edit?tab=t.0) provides some background and rationale. Note that the new `ancillaryInfo` idea is the most novel / interesting part of this proposal.  It has a basis in use cases collected from earlier landscape analyses, and perfectly addresses a key need for MC Statements -  but there are many alternative ways we might do this.  Worth a discussion at some point.

larrybabb committed:
update sm & github_version references in rst files

larrybabb committed:
added VarPath and VarOnco reportedIn, strength constraints for community versions

larrybabb committed:
fixed several doc links

larrybabb committed:
addition doc repairs and link fixes

larrybabb committed:
Sync doc changes (#301)

* additional doc updates to prep for forthcoming Mar 2025 ballot review.
---------
Authored-by: Matthew Brush <mhb120@gmail.com>

larrybabb committed:
Final Sync doc changes (#302) before Mar.2025 pre-release v1

* more cleanup and alignment with docs refactoring needed to prep Mar.2025 pre-release ahead of PRC review
* loosen strictness on acmg varpath reportedIn requirement for now
* added implementations table

---------
Co-authored-by: Matthew Brush <mhb120@gmail.com>

mbrush committed:
Add AAC constraints to therapeutic-response-study-statement-profile-s… (#304)

Added structure and descriptions needed to include AAC-based constraints in this profile, as we agreed on the 3-12-25 call.  
* Update therapeutic-response-study-statement-profile-source.yaml, diagnostic-study-statement-profile-source.yaml, and prognostic-study-statement-profile-source.yaml
* update cat-vrs sm to pull in vrs 2.0.1 release
* Altered approach to extending acmg, aac and ccv statement and evidLine profiles to enable more accurate alignment with community guidelines and extensions in general use
* remove CIViC SOP from AAC profiles
* cleanup oncogenicity descriptions [minor]
* cleanup oncogenicity functional EL descriptions [minor]
* cleanup pathogenicity descriptions [minor]
* update CIViC EIDs to use generic statement model

---------

Co-authored-by: Lawrence Babb <lbabb@broadinstitute.org>
Co-authored-by: Kori Kuzma <korikuzma@gmail.com>
Co-authored-by: Alex H. Wagner, PhD <Alex.Wagner@nationwidechildrens.org>

korikuzma committed:
fix: `membershipOperator` and `conditions`/`therapies` should be required (#309)

* Indentation was off

korikuzma committed:
fix: resolve typos (#310)

korikuzma committed:
docs: add more info on VA-Spec Python (#311)

close #305 

* Once we have Jupyter notebooks, we can update the docs

korikuzma committed:
chore: bump va-spec ballot version (#312)

* `1.0.0-ballot.2025-03.2` -> `1.0.0-ballot.2025-03.3`

mbrush committed:
Update profiling-methodology.rst

larrybabb committed:
Docs update v2 (#314)

Significant document (and image) updates to the community profiles, domain entities, examples, modeling foundations and profiling methodology sections
* refactored dynamic links to github artifacts to assure they work post-build with the correct branch and version
---------

Co-authored-by: Matthew Brush <mhb120@gmail.com>

larrybabb committed:
repaired link

larrybabb committed:
1.0 doc ballot v3 updates (#315)

- more technical, developer-focused content into a dedicated Developer Guide section of the docs. 
- moved the Modeling Foundations content that addresses many of the higher level questions from the PRC to be more prominent.

---------

Authored-by: Matthew Brush <mhb120@gmail.com>

larrybabb committed:
Issue 317 (#321)

* rename all `subtype` attributes to `<class>Type`
* change Onco/PathFunctImpactEvidenceLine to Onco/PathEvidenceLine in both ACMG and CCV community guideline profiles.
* add ALL acmg/ccv criterion codes as methodType enumberations to community evidenceLines.

larrybabb committed:
Fix: Formalized described conditional validation in json schema IF-THEN (#322)

mbrush committed:
Schema 'description' and 'comment' updates (#323)

Improving `descriptions` and `comments` across classes and profiles in
the specification after recent changes to better support ACMG based
profiles.

korikuzma committed:
feat!: revert caf.focusAllele change + add TumorVariantFrequencyStudy… (#320)

…Result

close #319

---------

Co-authored-by: Alex H. Wagner, PhD <Alex.Wagner@nationwidechildrens.org>
Co-authored-by: Matthew Brush <mhb120@gmail.com>

korikuzma committed:
chore: update va-spec tags to 1.0.0-ballot.2025-03.4 (#325)

Co-authored-by: Alex H. Wagner, PhD <Alex.Wagner@nationwidechildrens.org>

larrybabb committed:
1.0 docs ballot v3 (#327)

Authored-by: Matthew Brush <mhb120@gmail.com> based on post-hackathon updates and general feedback for improved organization.

larrybabb committed:
fix broken dynamic links (#332)

resolution to issue #330

larrybabb committed:
testing rtd (#333)

larrybabb committed:
resolve issue #330 (#335)

test RTD conf.py to capture release based on tag if available otherwise
git branch name

mbrush committed:
Added guidance for populating alleleOriginQualifier (#339)

- Added detail and considerations for populating the alleleOriginQualifier attribute - per feedback from the PRC.

mbrush committed:
Update proposition predicates with constant values (#338)

- added a "const" element to declare the constant value of predicates that currently have a single 'default' value
- also added text to descriptions of predicates that indicate permissible values, so they show up in web docs rendered by MSP.

larrybabb committed:
1.0 docs ballot v4 (#341)

Authored-by: Matthew Brush <mhb120@gmail.com>

mbrush committed:
Update index.rst

mbrush committed:
Update introduction.rst

mbrush committed:
Rename index.rst to index.rst

mbrush committed:
Rename acmg-2015.rst to acmg-2015.rst

mbrush committed:
Rename figure-formatting-test.rst to x-figure-formatting-test.rst

mbrush committed:
Update index.rst

mbrush committed:
Update core-classes.rst

mbrush committed:
Update data-structures.rst

mbrush committed:
Update README.md

larrybabb committed:
clean up whitespace issues

larrybabb committed:
1.0 docs ballot v5 (#344)

@mbrush documentation updates to clarify the two profiling approaches
available in va-spec.

---------

Authored-by: Matthew Brush <mhb120@gmail.com>

larrybabb committed:
update schema and associated doc link ids to match new pre-release ve… (#345)

@mbrush final doc updates in preparation for the PSC meeting. mostly
involves updates around the differences in the 2 profiling approaches.

mbrush committed:
Adding classes and profiles for representing molecular consequence

added VariantMolecularConsequenceProposition class to va-core-source.yaml

mbrush committed:
Update va-core-source.yaml

Adding an ancillaryInfo attribute to the core Statement class, to capture additional info that Epic wans to provide as part of an MC statement.

mbrush committed:
Create vep-transcript-consequence-statement.yaml

Adding a test fixture example for a transcript consequence

mbrush committed:
Create vep-intergenic-consequence-statement.yaml

Adding test fixtures example of an intergenic consequence statement

mbrush committed:
Rename vep-intergenic-consequence-statement.yaml to tests/fixtures/vep-intergenic-consequence-statement.yaml

moved example into right directory

mbrush committed:
Update vep-transcript-consequence-statement.yaml

updated objectSequence -> objectMolecule

mbrush committed:
Update va-core-source.yaml

updated objectSequence -> objectMolecule

mbrush committed:
Update vep-transcript-consequence-statement.yaml

minor fixes

mbrush committed:
Update vep-intergenic-consequence-statement.yaml

minor fixes

sarahhunt commented on schema/va-spec/base/va-core-source.yaml:
```diff
@@ -970,6 +983,63 @@ $defs:
+    maturity: draft
+    type: object
+    description: >-
+      A Proposition describing a type of consequence that a variant has on genomic, transcript, or polypeptide
```
has -> is predicted to have

add regulatory elements

sarahhunt commented on tests/fixtures/vep-transcript-consequence-statement.yaml:
```diff
@@ -0,0 +1,57 @@
+      type: SequenceReference
+      refgetAccession: SQ.Y2M5OWI4OGRiN2JkNGViYzQ2MjllMThk    
+      residueAlphabet: na
+    conseequenceTypeQualifiers:
```
conseequenceTypeQualifiers -> consequenceTypeQualifiers

mbrush replied:
thanks Sarah. I just fixed this typo.

Mrinal-Thomas-Epic commented on tests/fixtures/vep-intergenic-consequence-statement.yaml:
```diff
@@ -0,0 +1,21 @@
+    refgetAccession: SQ.Z04NDAWITHJ9GNBShi789hjh79Ag       # temp value - replace with true refget accession
+    residueAlphabet: na
+	conseequenceTypeQualifiers:
+    - name: intergenic_variant
```
Very minor, but I think the indentation is slightly off on this array element (which changes the meaning in YAML). You can try adding these two fixtures to `tests/test_definitions.yaml`, and it should throw errors in places where it doesn't validate against the JSON schema. I can also set this up and commit it to the PR if it helps

mbrush replied:
thanks Mrinal - not sure how it got so wonky.  just fixed it.

Mrinal-Thomas-Epic commented on schema/va-spec/base/va-core-source.yaml:
```diff
@@ -519,6 +519,19 @@ $defs:
+        An object in which a data provider can define custom attributes to capture additional information about the 
+        entities represented by the subject, object, or qualifiers of the Proposition asserted or assessed by a Statement.
+        This information can aid in understanding the significance or utility of the primary knowledge a Statement reports. 
+        For example, ancillary info for a Molecular Consequence Statement may include other transcript- or protein-level
+        representations of the variant, transcript or protein molecules on which the consequence applies, genes affected
```
Rather than putting it in the description, can you define a schema for `MolecularConsequenceAncillaryInfo`?

mbrush replied:
I certainly did consider this option when drafting the proposed model, and even made a comment about it in the google doc - which I will copy in below as well. Bottom line - we could choose to go this route, but might start with a minimal generic object for now and learn from how adopters populate it in their implementations.  Let's discuss when we talk next.

From the gdoc:

> Note that the initial proposal does not define a class with dedicated properties as the value of the `ancillaryInfo` attribute (e.g. a `VariationAncillaryInfo` class, or even a `MolecularConsequenceAncillaryInfo` class). It takes a generic object in which implementers can define the properties they need.

> We could however choose to define such specific ancillary info classes, which could include named attributes for the types of ancillary info VEP data provides around a MC annotation. e.g.:

````
VariationAncillaryInfo / MCAncillaryInfo:
  transcriptVariant: Allele | iriReference [0..1]
  transcriptMolecule: SequenceReference | iriReference [0..1]
  proteinVariant: Allele | iriReference [0..1]
  proteinMolecule: SequenceReference | iriReference [0..1]
  affectedGene: MappableConcept | iriReference [0..1]
  distanceToAffectedGene: number [0..1]
````

mbrush committed:
Update va-core-source.yaml

updated description of the MC Proposition class  to say 'predicted to have' vs 'has'

mbrush committed:
Update vep-intergenic-consequence-statement.yaml

fix yaml indent issues

mbrush committed:
Update vep-transcript-consequence-statement.yaml

fix typo

mbrush committed:
Update va-core-source.yaml

mbrush committed:
Update vep-transcript-consequence-statement.yaml

mbrush committed:
Update va-core-source.yaml

mbrush committed:
Update va-core-source.yaml

mbrush commented:
@ahwagner @larrybabb - this is my original PR for a MC Profile model - which is now obsolete given that the model we aligned on was implemented from Mrinal's PR #336.  

**HOWEVER**, a lot of commits that are unrelated to my MC proposal seem to have been added by various folks to this PR branch - which are things I suspect we do want to get into the v1 branch.  

Not sure why/how these features got added to this MC PR branch, but we should figure out how to untangle what additions we want to merge into the v1 branch, from those we don't (because they are superseded by the MC work in PR #336)

mbrush commented:
Closing and deferring to proposal in PR #336
